src/ssl.c

```diff
@@ -14275,7 +14282,14 @@ static int GetDeepCopySession(WOLFSSL* ssl, WOLFSSL_SESSION* copyFrom)
 
 int SetSession(WOLFSSL* ssl, WOLFSSL_SESSION* session)
 {
-    if (ssl == NULL || ssl->options.sessionCacheOff)
+    if (ssl == NULL || (ssl->options.sessionCacheOff
+#if defined(HAVE_SESSION_TICKET) && defined(WOLFSSL_FORCE_CACHE_ON_TICKET)
```

Comment:
Since we do this multiple times and it is messy, create a function or macro that tells whether the session cache is really off for this session.

Reply:
Refactored into `SslSessionCacheOn`.